docs: oauth2 provider migration (#5271)
strongloop/loopback-next: docs/site/migration/auth/oauth2.md

permalink: /doc/en/lb4/migration-auth-oauth2.html
---

[`loopback-component-oauth2`](https://github.com/strongloop/loopback-component-oauth2)
is used to setup a LoopBack application as an [OAuth 2.0](https://oauth.net/2/)
server. It can be applied when people implement their LoopBack application as an
OAuth 2.0 provider. For example, the API gateway application.

> At this stage, authentication and authorization in LoopBack 4 focus on the

Reviewer: > focus on the

focuses on the

Reviewer: @jannyHou , you are right `focuses` is fine.

> At this stage, authentication and authorization in LoopBack 4 focus on the

Reviewer 2: ```suggestion
As of now, LoopBack 4 focuses on redirecting to third-party OAuth2 providers like google or facebook for authentication and authorization, instead of turning itself into an OAuth 2.0 provider.
```

> like [google](https://developers.google.com/identity/protocols/oauth2) or

Reviewer 3: nitpick: `google` -> `Google`

> [facebook](https://developers.facebook.com/docs/facebook-login/) for

Reviewer 3: nitpick: `facebook` -> `Facebook`

> At this stage, authentication and authorization in LoopBack 4 focus on the
> communication to third-party servers like
> [google](https://developers.google.com/identity/protocols/oauth2) or
> [facebook](https://developers.facebook.com/docs/facebook-login/), instead of
> turning itself into an OAuth 2.0 provider. And given the complexity of

Reviewer 2: ```suggestion
As of now, LoopBack 4 focuses on redirecting to third-party OAuth2 providers like google or facebook for authentication and authorization, instead of turning itself into an OAuth 2.0 provider.
```

[`loopback-component-oauth2`](https://github.com/strongloop/loopback-component-oauth2),

> we decide to postpone the migration guide to next stage.

Reviewer 2: ```suggestion
we have decided to defer the migration guide for `loopback-component-oauth2`.
```

We have a simplified OAuth 2.0 server implementation in example
[passport-login](https://github.com/strongloop/loopback-next/tree/master/examples/passport-login),

> which sets up an express application as an OAuth 2.0 provider. The provider is

Reviewer: > The provider is created in file mock-oauth2-social-app.ts

Do you mean we `also` have another example of an oauth provider in mock-oauth2-social-app.ts ?

The way it is currently worded, it makes it sound as if mock-oauth2-social-app.ts is being used in examples/passport-login .

Author: > mock-oauth2-social-app.ts is being used in examples/passport-login 

 @emonddr It is a fixture in the example's test folder. Hmm, is the meaning clear now?

> test fixtures, which sets up an express application as an OAuth 2.0 provider.

Reviewer 3: nitpick: `express` -> `Express`

created in file
[mock-oauth2-social-app.ts](https://github.com/strongloop/loopback-next/blob/master/extensions/authentication-passport/src/__tests__/acceptance/fixtures/mock-oauth2-social-app.ts).

> At this moment people who are interested to create such a provider can use it as

Reviewer: > create such a provider can use it as

create such  providers can use them as

(this sentence only needs to change if we are talking about 2 distinct examples of an oauth provider)

Author: oh there is only one provider. Let me rephrase the sentence above.
